### PR TITLE
fix: some python hook fixes and tests

### PR DESCRIPTION
**Pull request by mkenigs**
- Made some changes I thought were non-controversial
- Added some tests
  - test_get_init_customization_with_selected_provider and test_get_init_customization_with_providers are unfinished
  - test_poetry_pyproject_unavailable_version and test_pyproject_unavailable_version have legitimate failures that need to be fixed

### CODE REVIEW

File: cli/flox/src/commands/init/python.rs [hunk 1]
```diff
@@ -662,3 +671,236 @@ impl Display for ProvidedVersion {
...
+            selected_provider: Some(Box::new(PoetryPyProject {
+                provided_python_version: ProvidedVersion::Compatible {
+                    requested: None,
+                    compatible: "3.11.6".to_string(),
+                },
+                poetry_version: "1.7.1".to_string(),
+            })),
+        };
+        assert_eq!(python.get_init_customization(), todo!());
+    }
```
Comment (mkenigs) [inline cli/flox/src/commands/init/python.rs:735]: Could maybe drop these in favor of an e2e test

File: cli/flox/src/commands/init/python.rs [hunk 2]
```diff
@@ -662,3 +671,236 @@ impl Display for ProvidedVersion {
...
+        "#};
+
+        let pyproject = PoetryPyProject::from_pyproject_content(content, flox);
+
+        assert!(pyproject.is_err());
+    }
+
+    /// ProvidedVersion::Compatible should be returned for python = "^3.7"
+    #[test]
+    #[serial]
```
Comment (ysndr) [inline cli/flox/src/commands/init/python.rs:859]: why do some of these have to be serial?
related to pkgdb scrapes?

File: cli/flox/src/commands/init/python.rs [hunk 3]
```diff
@@ -662,3 +671,236 @@ impl Display for ProvidedVersion {
...
+    fn test_poetry_pyproject_available_version() {
+        let flox = &FLOX_INSTANCE.0;
+
+        // TODO: python docs have a space in the version (>= 3.8)
+        // https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#python-requires
+        // pkgdb currently throws an exception when passed that specifier
+        let content = indoc! {r#"
+        [tool.poetry.dependencies]
+        python = "^3.7"
+        "#};
```
Comment (ysndr) [inline cli/flox/src/commands/init/python.rs:869]: poetry seem to adhere to standard semver: https://python-poetry.org/docs/dependency-specification#caret-requirements

though they might be more lenient with it so fixing it for both cases seems reasonable anyway

**Comment by ysndr** [comment]
Thanks @mkenigs I'll continue working on these on the original branch

File: cli/flox/src/commands/init/python.rs (hunk 1 above)
Comment (ysndr) [reply cli/flox/src/commands/init/python.rs:735]: wdym, shall we adapt the extend upon the `"'flox init' sets up a working Python environment using requirements.txt"` bats test instead?

I added bats tests in https://github.com/flox/flox/pull/1141 lmk if we also want to test verbatim output of these abstract hooks.

**Pull request merged**